### ISSUE TITLE
Extension labeling and versioning in Metadata

### ISSUE THREAD

**Issue by pagbabian-splunk** [ocsf/ocsf-schema#474]
There is no way within an event to determine whether an extension is part of the event, unless the extension is a net-new event class (or possibly Profile etc) with new names that indicate who has extended the schema.  For example, Splunk had a Network File Activity extension class, and a Detection Report extension class which would not have been obvious as to who extended the schema.  Even with a descriptive name indicating the extender, there may be more than one version of the extension.

At the expense of further bloat of the event, I'm suggesting that Metadata more explicitly have a place to indicate the producer or consumer that has extended that portion of the schema, and their version.  The version may in fact be more important than the vendor, since just an event name with vendor prefix, for example, would not convey what version.

**Comment by jasonbreimer** [comment]
Yes, agreed. Going back to existing extensions makes sense. As you stated version would be very important too. +1 vote.

{
  "caption": "New or Existing Extension",
  "meta": "profile",
  "name": "name_ex",
  "version": "0.0.0",
  "uid": 123
}

**Comment by pagbabian-splunk** [comment]
Thanks Jason - I will convert this issue to a Proposal
